parse_number_amd64.go

```diff
@@ -38,6 +39,12 @@ func parse_number_simd(buf []byte, found_minus bool) (success, is_double bool, d
 		}
 
 		var err error
+		// check if number would overflow
+		_, err = strconv.ParseUint(string(buf[:pos]), 10, 64)
```

Comment:
This seems to prevent big numbers to be returned as floats.

These 2 unit tests fail when this is in place:

https://github.com/karlmcguire/simdjson-go/blob/e09093318f0c57c1bf817744ca1e823d1f29011a/simdjson_amd64_test.go#L933-L946

So this cannot be used as-is. I will take a look at this piece of code and see what can be done.